packages/fx-core/src/common/local/localEnvManager.ts

```diff
@@ -19,28 +19,43 @@ import { LocalCrypto } from "../../core/crypto";
 import { CoreSource, ReadFileError } from "../../core/error";
 
 class LocalEnvManager {
-  public async getLaunchInput() {}
+  public getLaunchInput(rawLocalSettings: Json | undefined): any {
+    // return local teams app id
+    const localTeamsAppId = rawLocalSettings?.teamsApp?.teamsAppId as string;
+    return { appId: localTeamsAppId };
+  }
 
   public async getLocalDebugEnvs() {}
 
   public async getNpmInstallLogInfo() {}
 
   public async getPortsInUse() {}
 
-  public async getProgrammingLanguage() {}
+  public getProgrammingLanguage(projectSettings: ProjectSettings): string | undefined {
+    return projectSettings.programmingLanguage;
+  }
 
-  public async getSkipNgrokConfig() {}
+  public getSkipNgrokConfig(rawLocalSettings: Json | undefined): boolean {
+    return (rawLocalSettings?.bot?.skipNgrok as boolean) === true;
+  }
 
-  private async getLocalSettings(projectPath: string): Promise<Json | undefined> {
-    const projectSettings = await this.getProjectSettings(projectPath);
+  public async getLocalSettings(projectPath: string, projectId: string): Promise<Json | undefined> {
```

Comment:
Why not use `crypto` instead of `projectId` as parameter, then `getLocalSettings` and `getRawLocalSettings` can be merged into one function?

Reply:
Good catch. Merged into one func but expose `cryptoOption` as parameter. (`CryptoProvider` is core internal so do not want to be exported to cli/vsc)